test/unit/realm_test.rb

```diff
@@ -8,42 +8,30 @@ def setup
   end
 
   test "should not save without a name" do
-    assert !@new_realm.save
+    assert validate_presence_of(:name)
   end
 
   test "should exists a unique name" do
-    other_realm = Realm.new(:name => "myrealm.net")
-    assert !other_realm.save
+    assert validate_uniqueness_of(:name)
   end
 
   test "when cast to string should return the name" do
-    s = @realm.to_s
-    assert_equal @realm.name, s
+    assert_equal @realm.name, @realm.to_s
   end
 
   test "should not destroy if it contains hosts" do
-    disable_orchestration
-    host = FactoryGirl.create(:host, :realm => @realm)
-
-    assert host.save
-
-    realm = host.realm
-    assert !realm.destroy
-    assert_match /is used by/, realm.errors.full_messages.join("\n")
+    assert have_many(:hosts).
+      class_name('Host::Managed').
+      dependent(:destroy)
```

Comment:
This looks wrong, why is it passing?  Realms should *not* be destroyed when hosts are removed, or vice versa.